packages/workspace/src/browser/workspace-uri-contribution.spec.ts

```diff
@@ -14,58 +14,178 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
+import { Container, ContainerModule } from 'inversify';
+import { FileStat, FileSystem } from '@theia/filesystem/lib/common/filesystem';
+import { MockFilesystem } from '@theia/filesystem/lib/common/test';
+import { FOLDER_ICON, FILE_ICON, DefaultUriLabelProviderContribution } from '@theia/core/lib/browser/label-provider';
+import { IWorkspaceService } from './workspace-service';
 import { WorkspaceUriLabelProviderContribution } from './workspace-uri-contribution';
-import { Container, ContainerModule, injectable } from 'inversify';
+import { MockWorkspaceService } from '../common/test/mock-workspace-service';
 import URI from '@theia/core/lib/common/uri';
-import { IWorkspaceService } from './workspace-service';
-import { FileStat, FileSystem } from '@theia/filesystem/lib/common/filesystem';
 import { expect } from 'chai';
-import { MockFilesystem } from '@theia/filesystem/lib/common/test';
+import * as sinon from 'sinon';
 
+let container: Container;
 let labelProvider: WorkspaceUriLabelProviderContribution;
-
-@injectable()
-class MockWorkspaceService implements IWorkspaceService {
-    get roots(): Promise<FileStat[]> {
-        const stat: FileStat = {
-            uri: 'file:///workspace',
-            lastModification: 0,
-            isDirectory: true,
-        };
-        return Promise.resolve([stat]);
-    }
-}
-
-beforeEach(function() {
-
+before(() => {
     const module = new ContainerModule(bind => {
         bind(WorkspaceUriLabelProviderContribution).toSelf().inSingletonScope();
         bind(IWorkspaceService).to(MockWorkspaceService).inSingletonScope();
-        bind(FileSystem).to(MockFilesystem);
+        bind(FileSystem).to(MockFilesystem).inSingletonScope();
     });
-    const container = new Container();
+    container = new Container();
     container.load(module);
```

Comment:
You can create a container and directly use `bind` on it.

Reply:
fixed. thank you !